### PR TITLE
Updating curry definition/example

### PR DESCRIPTION
**Pull request by CrossEye**
This is more complex a definition, but I think it's more accurate.

### CODE REVIEW

**Comment by weblancaster** [comment]
I was able to understand better though. +1

File: readme.md [hunk 1]
```diff
@@ -50,14 +50,14 @@ partial(2); //=> 42
 
 ## Currying
 
-> The process of converting a function with multiple arity into the same function with an arity of one. Not to be confused with partial application, which can produce a function with an arity greater than one.
+> The process of converting a function of multiple parameters into one, which, when called with fewer parameters, returns a new function that awaits the remaining parameters.  It's important to note that this returned function, if there are still more than one parameter unspecified, is also curried.  Strictly speaking, it is only currying when the returned function accepts just one parameter, but many implementations allow for multiple parameters at each level.  This is not to be confused with partial application, in which the returned function still requires all remaining parameters to be supplied.
```
Comment (weblancaster) [inline readme.md]: I think adding `in which the returned function still requires all remaining parameters to be supplied.` just add more pieces to put together, so removing this sentence would be good.

Comment (CrossEye) [inline readme.md]: I think this version is a bit too wordy, and would like to see it streamlined.  But I would like to see most of this content included.

Any suggestions for how to distinguish this from partial application (which, I, for one, find important here)?

Comment (davidchambers) [inline readme.md]: _Currying is the **transformation** from a function with multiple arguments to a function which takes its arguments one at a time. If `f` is a ternary function and `g` is the result of currying `f`, `g(a)(b)(c)` is equivalent to `f(a, b, c)`._

Comment (CrossEye) [inline readme.md]: That's nice @davidchambers.  "**transformation**" was definitely the missing word.  But I still like capturing the notion that a lot of libraries have of a slightly more generic form of currying, where `g(a, b)(c)`, `g(a)(b, c)`, and `g(a, b, c)` are all equivalent to `g(a)(b)(c)`.

Comment (hemanth) [inline readme.md]: Nice definition @davidchambers :+1: 

@weblancaster You definition is good but as @CrossEye mentioned it's too word, we would need a TLDR of that.

@CrossEye There is this thin line between `partial application` and `currying` that is tricky at times to get noticed.

Comment (CrossEye) [inline readme.md]: FTR, the wordy definition was mine.  I don't know the overall plan for this document, but I might suggest that the best bet would be to have a brief definition that captures the essence and then if necessary a few additional sentences which clarify it, add caveats, whatever.  (And of course an example or two to follow that.)  With that said, I would nominate the definition from @davidchambers as the key definition, but would back it up, perhaps with something like this:

> _Currying is the transformation from a function with multiple arguments to a function which takes its arguments one at a time. If `f` is a ternary function and `g` is the result of currying `f`, `g(a)(b)(c)` is equivalent to `f(a, b, c)`._
> 
> Some versions of currying allow for additional ways to call, so that these are all equivalent: `g(a, b, c)`, `g(a, b)(c)`, `g(a)(b, c)`, `g(a)(b)(c)`.  Currying is not to be confused with partial application:  If you partially apply the first parameter to `f`, you get a plain binary function.  If you curry `f` and apply the first parameter, you get a new _curried_ function.  That is, `g(a)` is also a curried function. 
> 
> ```js
> let sum = (a, b, c) =>  a + b + c;
> let curriedSum = a => b => c => a + b + c;
> curriedSum(15)(25)(2) //=> 42
> ```

Comment (hemanth) [inline readme.md]: Awesomeness++ :+1:

Comment (davidchambers) [inline readme.md]: > If you partially apply the first parameter to `f`, you get a plain binary function.

Does it make more sense to discuss partial application in the context of curried functions? Given

```javascript
// f :: *,*,* -> *
const f = (a, b, c) => …;
```

and

```javascript
// g :: * -> * -> * -> *
const g = curry(f);
```

we could simply use

```javascript
g(x)  // :: * -> * -> *
```

as an example of partial application rather than

```javascript
partial(f, x)  // :: *,* -> *
```

which would need an accompanying explanation of `partial`.

We could then mention that it is possible to partially apply a function which expects multiple arguments at once, but that this requires the use of a function written specifically for this purpose.

Comment (hemanth) [inline readme.md]: > Does it make more sense to discuss partial application in the context of curried functions?

^ Yes! I got the same feedback from few others as well.

Comment (jayfunk) [inline readme.md]: @CrossEye Do you want to update your commit to match @davidchambers updated definition? That way this is basically ready for merging.
